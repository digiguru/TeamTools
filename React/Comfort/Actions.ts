export const ComfortActions = {
    SET_STAGESIZE : "SET_STAGESIZE",
    SET_ZONEFOCUS : "SET_ZONEFOCUS",
    SET_USERFOCUS : "SET_USERFOCUS",
    SELECT_USER   : "SELECT_USER",
    CHOOSE_ZONE   : "CHOOSE_ZONE",
    TOGGLE_CHOICES: "TOGGLE_CHOICES"
};

export function setStageSize(width: number, height: number) {
    return {type: ComfortActions.SET_STAGESIZE, width: width, height: height};
}
export function setUserFocus(user: string, focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_USERFOCUS, user: user, focus: focus};
}
export function setZoneFocus(area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus") {
    return {type: ComfortActions.SET_ZONEFOCUS, area: area, focus: focus};
}
export function selectUser(user: string) {
    return {type: ComfortActions.SELECT_USER, user: user};
}
export function chooseZone(user: string, area: "Chaos" | "Stretch" | "Comfort", distance: number) {
    return {type: ComfortActions.CHOOSE_ZONE, user: user, area: area, distance: distance};
}
export function toggleChoiceVisibility(visible: boolean) {
    return {type: ComfortActions.TOGGLE_CHOICES, visible: visible};
}
