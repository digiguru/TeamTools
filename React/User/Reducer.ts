export function userReducer(state: any, action): any {
    return {};
}