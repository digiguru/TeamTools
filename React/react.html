<html>
    <head>

           <link rel="stylesheet" href="../Shared/styles.css">

    </head> 
        
    
    <body id="stage" class="coloured-bg">
        <div id="tuckman"></div> 
        <div id="container"></div>
        <div id="comfort"></div>

    </body>
    <script src="../3rdParty/require.min.js" data-main="js/compiled"></script>
</html>