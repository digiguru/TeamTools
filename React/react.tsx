/*
<ul id="users"><li><span class="user">Adam Hall</span><a href="void(0);">X</a></li><li><span class="user">Billie Davey</span><a href="void(0);">X</a></li><li><span class="user">Laura Rowe</span><a href="void(0);">X</a></li><li><span class="user">Simon Dawson</span><a href="void(0);">X</a></li><li><span class="user">Fred</span><a href="void(0);">X</a></li></ul>
*/
class Delete extends React.Component<any, any> {
    render() {
        return <a href="void(0);">X</a>;
    }
}
class User extends React.Component<any, any> {
  render() {
    return <li><span className="user">{this.props.username}</span><Delete /></li>;
  }
}
class UserList extends React.Component<any, any> {
    render() {
        let users = [];

        this.props.users.forEach(function(user) {
            users.push(<User user={user} username={user.name} key={user.name} />);
        });
        return <ul id="users">{users}</ul>;
    }
}
let USERS = [
    {name: "bob"},
    {name: "Donald"}
];
ReactDOM.render(
  <UserList users={USERS} />,
  document.getElementById("container")
);