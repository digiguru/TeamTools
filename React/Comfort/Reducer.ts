
import {ComfortActions} from "./Actions";
import {ComfortAppState} from "./Model";
import { fromJS, List, Map } from "immutable";
import { Point } from "../Models/Point";
import { Size } from "../Models/Size";
import { DOMMeasurement } from "../Models/IDomMeasurement";


const initialSize: Size = new Size(800, 800);
const initialState: ComfortAppState = {
    Size: initialSize,
    CenterPoint: new Point(initialSize.width / 2, initialSize.height / 2),
    UserList : {
        ShowUsers: true,
        Users: [
            {Username: "Adam Hall", Focus: "not-in-focus", Y: 0 },
            {Username: "Caroline Hall", Focus: "not-in-focus", Y: 0 }
        ]
    },
    Zones : {
        Comfort: {Name: "Comfort", Focus: "not-in-focus", Range: {Start: 0, End: 100}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Stretch: {Name: "Stretch", Focus: "not-in-focus", Range: {Start: 100, End: 200}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Chaos: {Name: "Chaos", Focus: "not-in-focus", Range: {Start: 200, End: 300}, Size: {Width: new DOMMeasurement("100%"), Height: new DOMMeasurement("100%")}}
    },
    ShowUserChoices: false,
    UserChoices: []
};
export function comfortReactApp(state: ComfortAppState = initialState, action): ComfortAppState {
    console.log(action.type, action);
    switch (action.type) {
        case ComfortActions.SET_STAGESIZE:
            return ComfortZoneAction.setStageSize(state, (<any>action).width, (<any>action).height);
        case ComfortActions.SET_USERFOCUS:
            return ComfortZoneAction.setUserFocus(state, (<any>action).user, (<any>action).focus);
        case ComfortActions.SET_ZONEFOCUS:
            return ComfortZoneAction.setZoneFocus(state, (<any>action).area, (<any>action).focus);
        case ComfortActions.SELECT_USER:
            return ComfortZoneAction.selectUser(state, (<any>action).user);
        case ComfortActions.CHOOSE_ZONE:
            return ComfortZoneAction.chooseZone(state, (<any>action).user, (<any>action).area, (<any>action).distance, (<any>action).x, (<any>action).y);
        case ComfortActions.TOGGLE_CHOICES:
            return ComfortZoneAction.toggleChoiceVisibility(state, (<any>action).visible);
        default:
            return state;
    }
}
class ComfortZoneAction {
    static setStageSize(state: ComfortAppState, width: number, height: number) {
        const newCenter = new Point(width / 2, height / 2);
        return fromJS(state)
            .set("Size", new Size(width, height))
            .set("CenterPoint", newCenter)
            .toJS();
    }
    static setZoneFocus(state: ComfortAppState, area: "Chaos" | "Stretch" | "Comfort", focus: "in-focus" | "active" | "not-in-focus"): ComfortAppState {
        return fromJS(state)
            .setIn(["Zones", "Comfort", "Focus"], area === "Comfort" ? focus : "not-in-focus")
            .setIn(["Zones", "Stretch", "Focus"], area === "Stretch"  ? focus : "not-in-focus")
            .setIn(["Zones", "Chaos", "Focus"], area === "Chaos"  ? focus : "not-in-focus").toJS();
    }
    static setUserFocus(state: ComfortAppState, user: string, focus: "in-focus" | "active" | "not-in-focus"): ComfortAppState {
        const originalList = List(state.UserList.Users);
        const newUserList = originalList.update(
            originalList.findIndex(item => item.Username === user),
            (item) => fromJS(item).set("Focus", focus)).toJS();
        return fromJS(state)
            .setIn(["UserList", "Users"], newUserList).toJS();
    }
    static selectUser(state: ComfortAppState, user: String): ComfortAppState {
        // Sets currentUser, and therefor hides the user choice menu
        const data = fromJS(state)
            .set("CurrentUser", user)
            .set("ShowUserChoices", false)
            .setIn(["UserList", "ShowUsers"], false);

        return data.toJS();
    }

    static chooseZone(state: ComfortAppState, user: string, area: "Chaos" | "Stretch" | "Comfort", distance: number, x: number, y:number): ComfortAppState {

        // Add the user choice
        const newUserChoices = List(state.UserChoices).push({
            User: user,
            Zone: area,
            Distance: distance
        }).toJS();
        // Remove the user from the choice list
        const newUserList = List(state.UserList.Users).filter((item) => item.Username !== user).toArray();
        // Show the user list
        const showUserChoice = !!(newUserList.length);
        // Return
        return fromJS(state)
            .delete("CurrentUser")
            .set("CenterPoint", new Point(x,y))
            .set("ShowUserChoices", showUserChoice)
            .set("UserChoices", newUserChoices)
            .setIn(["UserList", "Users"], newUserList)
            .setIn(["UserList", "ShowUsers"], showUserChoice).toJS();
    }
    static toggleChoiceVisibility(state: ComfortAppState, visible: boolean): ComfortAppState {
        // Set "showUserChoices" to true
        return Map(state)
            .set("ShowUserChoices", visible).toJS();
    };
}