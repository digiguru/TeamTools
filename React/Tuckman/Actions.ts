import { IUserList } from "../User/Model";

export const TuckmanActions = {
    SET_STAGESIZE : "SET_STAGESIZE",
    SET_ZONEFOCUS : "SET_ZONEFOCUS",
    SET_USERFOCUS : "SET_USERFOCUS",
    SET_USERLIST  : "SET_USERLIST",
    SELECT_USER   : "SELECT_USER",
    CHOOSE_ZONE   : "CHOOSE_ZONE",
    TOGGLE_CHOICES: "TOGGLE_CHOICES"
};

export function setStageSize(width: number, height: number) {
    return {type: TuckmanActions.SET_STAGESIZE, width: width, height: height};
}
export function setZoneFocus(area: "forming" | "storming" | "norming" | "performing", focus: "in-focus" | "active" | "not-in-focus") {
    return {type: TuckmanActions.SET_ZONEFOCUS, area: area, focus: focus};
}

export function setUserFocus(user: string, focus: "in-focus" | "active" | "not-in-focus") {
    return {type: TuckmanActions.SET_USERFOCUS, user: user, focus: focus};
}
export function setUserList(userList: IUserList) {
    return {type: TuckmanActions.SET_USERLIST, userList: userList};
}
export function selectUser(user: string) {
    return {type: TuckmanActions.SELECT_USER, user: user};
}
export function chooseZone(user: string, area: "forming" | "storming" | "norming" | "performing", distance: number) {
    return {type: TuckmanActions.CHOOSE_ZONE, user: user, area: area, distance: distance};
}
export function toggleChoiceVisibility(visible: boolean) {
    return {type: TuckmanActions.TOGGLE_CHOICES, visible: visible};
}

