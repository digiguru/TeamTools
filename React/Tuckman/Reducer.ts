
// import {ComfortActions} from "./Actions";
import { ITuckmanModel, ITuckmanZoneList } from "./Model";
import { fromJS, List, Map } from "../3rdParty/immutable.min";
import { DOMMeasurement } from "../Models/IDomMeasurement";
import { Size } from "../Models/Size";
import { Point } from "../Models/Point";
import { TuckmanActions } from "./Actions";
import { IUser } from "../User/Model";


const initialSize: Size = new Size(800, 800);
const initialState: ITuckmanModel = {
    UserList : {
        ShowUsers: true,
        Users: [
            {Username: "Adam Hall", Focus: "not-in-focus", Y: 0 },
            {Username: "Caroline Hall", Focus: "not-in-focus", Y: 0 }
        ]
    },
    Size: initialSize,
    zones: {
        forming   : {index: 0, label: "forming",    focus: "not-in-focus", Events: undefined},
        storming  : {index: 1, label: "storming",   focus: "not-in-focus", Events: undefined},
        norming   : {index: 2, label: "norming",    focus: "not-in-focus", Events: undefined},
        performing: {index: 3, label: "performing", focus: "not-in-focus", Events: undefined}
    },
    UserChoices: []
    /*CenterPoint: new Point(initialSize.width / 2, initialSize.height / 2),

    Zones : {
        Comfort: {Name: "Comfort", Focus: "not-in-focus", Range: {Start: 0, End: 100}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Stretch: {Name: "Stretch", Focus: "not-in-focus", Range: {Start: 100, End: 200}, Size: {Width: new DOMMeasurement("50%"), Height: new DOMMeasurement("50%")}},
        Chaos: {Name: "Chaos", Focus: "not-in-focus", Range: {Start: 200, End: 300}, Size: {Width: new DOMMeasurement("100%"), Height: new DOMMeasurement("100%")}}
    },
    ShowUserChoices: false,
    */

};

export function tuckmanReactApp(state: ITuckmanModel = initialState, action): ITuckmanModel {
    switch (action.type) {
       case TuckmanActions.SET_STAGESIZE:
            return TuckmanZoneAction.setStageSize(state, (<any>action).width, (<any>action).height);
        case TuckmanActions.SET_USERFOCUS:
            return TuckmanZoneAction.setUserFocus(state, (<any>action).user, (<any>action).focus);
        case TuckmanActions.SET_ZONEFOCUS:
            return TuckmanZoneAction.setZoneFocus(state, (<any>action).area, (<any>action).focus);
         case TuckmanActions.SELECT_USER:
            return TuckmanZoneAction.selectUser(state, (<any>action).user);
        case TuckmanActions.CHOOSE_ZONE:
            return TuckmanZoneAction.chooseZone(state, (<any>action).user, (<any>action).area, (<any>action).distance);
        case TuckmanActions.TOGGLE_CHOICES:
            return TuckmanZoneAction.toggleChoiceVisibility(state, (<any>action).visible);
        default:
            return state;
    }
}
class TuckmanZoneAction {

    static setStageSize(state: ITuckmanModel, width: number, height: number): ITuckmanModel {
        const newCenter = new Point(width / 2, height / 2);
        return fromJS(state)
            .set("Size", new Size(width, height))
            .set("CenterPoint", newCenter)
            .toJS();
    }
    static setZoneFocus(state: ITuckmanModel, area: "forming" | "storming" | "norming" | "performing", focus: "in-focus" | "active" | "not-in-focus"): ITuckmanModel {
        return fromJS(state)
            .setIn(["zones", "forming", "focus"], area === "forming" ? focus : "not-in-focus")
            .setIn(["zones", "storming", "focus"], area === "storming"  ? focus : "not-in-focus")
            .setIn(["zones", "norming", "focus"], area === "norming"  ? focus : "not-in-focus")
            .setIn(["zones", "performing", "focus"], area === "performing"  ? focus : "not-in-focus").toJS();
    }
    static setUserFocus(state: ITuckmanModel, user: string, focus: "in-focus" | "active" | "not-in-focus"): ITuckmanModel {
        const originalList = List(state.UserList.Users);
        const newUserList = originalList.update(
            originalList.findIndex(item => item.Username === user),
            (item) => fromJS(item).set("Focus", focus)).toJS();
        return fromJS(state)
            .setIn(["UserList", "Users"], newUserList).toJS();
    }
    static selectUser(state: ITuckmanModel, user: String): ITuckmanModel {
        const originalList = List(state.UserList.Users);
        const item: IUser = originalList.find(item => item.Username === user);
        // Sets currentUser, and therefor hides the user choice menu
        const data = fromJS(state)
            .set("CurrentUser", item)
            .set("ShowUserChoices", false)
            .setIn(["UserList", "ShowUsers"], false);

        return data.toJS();
    }

    static chooseZone(
        state: ITuckmanModel, user: string, area: "forming" | "storming" | "norming" | "performing",
        distance: number): ITuckmanModel {

        // Add the user choice
        const newUserChoices = List(state.UserChoices).push({
            User: {Username: user},
            Zone: area,
            Distance: distance
        }).toJS();
        // Remove the user from the choice list
        const newUserList = List(state.UserList.Users).filter((item) => item.Username !== user).toArray();
        // Show the user list
        const showUserChoice = !!(newUserList.length);
        // Return
        return fromJS(state)
            .delete("CurrentUser")
            .set("ShowUserChoices", showUserChoice)
            .set("UserChoices", newUserChoices)
            .setIn(["UserList", "Users"], newUserList)
            .setIn(["UserList", "ShowUsers"], showUserChoice).toJS();
    }
    static toggleChoiceVisibility(state: ITuckmanModel, visible: boolean): ITuckmanModel {
        // Set "showUserChoices" to true
        return Map(state)
            .set("ShowUserChoices", visible).toJS();
    };
}
