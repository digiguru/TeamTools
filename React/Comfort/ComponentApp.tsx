import * as React from "react";
import { connect } from "react-redux";
import {BouncyAnimation, Events} from "../SVGHelper";
import {ChaosPickerState, ChaosPickerZoneState} from "./Model";
import { selectUser, chooseZone, toggleChoiceVisibility } from "./Actions";
import {ReduxChaosConnector, ReduxStretchConnector, ReduxComfortConnector} from "../ComfortZone/Connector";
import {ReduxUserConnector} from "../User/Connector";
import {ReduxUserHistoryConnector} from "../UserHistory/Connector";
import { ReduxUserHistory, ReduxUserHistoryArea } from "../UserHistory/Component";
import { StageConnector } from "../ComfortZone/Component";

export const ComfortApp = () => (
    <StageConnector>
        <ReduxChaosConnector Name="Chaos" />
        <ReduxStretchConnector Name="Stretch" />
        <ReduxComfortConnector Name="Comfort" />
        <ReduxUserConnector />
        <ReduxUserHistoryConnector />
    </StageConnector>
);
