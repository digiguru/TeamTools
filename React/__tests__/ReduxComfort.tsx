import * as React from "react";
import {createStore} from "../../3rdParty/redux.min";
import { ComfortApp } from "../Comfort/ComponentApp";
import {comfortReactApp} from "../Comfort/Reducer";
import { render } from "react-dom";
import { Provider } from "../../3rdParty/react-redux.min";
import * as Action from "../Comfort/Actions";

const renderizer = require("react-test-renderer");

it("Should not mutate in any way", () => {
    const myStore = createStore(comfortReactApp);
    const originalState = myStore.getState();
    const inputState = JSON.stringify(originalState);
    const checkAfterAction = (action) => {
        myStore.dispatch(action);
        expect(inputState).toEqual(JSON.stringify(originalState));
    };
    checkAfterAction(Action.setUserFocus("Adam Hall", "in-focus"));
    checkAfterAction(Action.selectUser("Adam Hall"));
    checkAfterAction(Action.setStageSize(800, 600));
    checkAfterAction(Action.setZoneFocus("Chaos", "in-focus"));
    checkAfterAction(Action.setZoneFocus("Stretch", "active"));
    checkAfterAction(Action.setZoneFocus("Comfort", "in-focus"));
    checkAfterAction(Action.toggleChoiceVisibility(true));
    checkAfterAction(Action.toggleChoiceVisibility(false));
    checkAfterAction(Action.chooseZone("Adam Hall", "Stretch", 85));
});



it("Should show the component", () => {
    // Arrange

    const myStore = createStore(comfortReactApp);
    const component = renderizer.create(
        <Provider store={myStore}>
            <ComfortApp />
        </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    myStore.dispatch(Action.setUserFocus("Adam Hall", "in-focus"));
    expect(component.toJSON()).toMatchSnapshot();

});

it("Should allow shrinking", () => {
    // Arrange

    const myStore = createStore(comfortReactApp);

    const component = renderizer.create(
        <Provider store={myStore}>
            <ComfortApp />
        </Provider>
    );
    myStore.dispatch(Action.chooseZone("Adam Hall", "Stretch", 50));
    expect(component.toJSON()).toMatchSnapshot();

    myStore.dispatch(Action.chooseZone("Caroline Hall", "Chaos", 100));
    expect(component.toJSON()).toMatchSnapshot();

});



