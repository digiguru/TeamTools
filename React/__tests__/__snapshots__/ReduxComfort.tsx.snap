exports[`test Should show the component 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <rect
      className="not-in-focus"
      height="100%"
      id="chaos"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      width="100%" />
    <text
      className="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20">
      chaos
    </text>
  </g>
  <g>
    <circle
      className="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      r="33%"
      width="50%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%">
      stretch
    </text>
  </g>
  <g>
    <circle
      className="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      r="15%"
      width="50%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%">
      comfort
    </text>
  </g>
  <g
    id="users">
    <g
      className="user-group">
      <rect
        className="not-in-focus"
        height="90"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        width="800"
        x="0"
        y={60} />
      <text
        className="username"
        x="60"
        y={60}>
        Adam Hall
      </text>
    </g>
    <g
      className="user-group">
      <rect
        className="not-in-focus"
        height="90"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        width="800"
        x="0"
        y={150} />
      <text
        className="username"
        x="60"
        y={150}>
        Caroline Hall
      </text>
    </g>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <rect
      class="not-in-focus"
      height="100%"
      id="chaos"
      width="100%"
    >
    </rect>
    <text
      class="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20"
    >
      chaos
    </text>
  </g>
  <g>
    <circle
      class="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      r="33%"
      width="50%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%"
    >
      stretch
    </text>
  </g>
  <g>
    <circle
      class="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      r="15%"
      width="50%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      comfort
    </text>
  </g>
  <g id="users">
    <g class="user-group">
      <rect
        class="not-in-focus"
        height="90"
        width="800"
        x="0"
      >
      </rect>
      <text
        class="username"
        x="60"
      >
        Adam Hall
      </text>
    </g>
    <g class="user-group">
      <rect
        class="not-in-focus"
        height="90"
        width="800"
        x="0"
      >
      </rect>
      <text
        class="username"
        x="60"
      >
        Caroline Hall
      </text>
    </g>
  </g>
</svg>
`;

exports[`test Should show the component 2`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g>
    <rect
      className="not-in-focus"
      height="100%"
      id="chaos"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      width="100%" />
    <text
      className="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20">
      chaos
    </text>
  </g>
  <g>
    <circle
      className="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      r="33%"
      width="50%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%">
      stretch
    </text>
  </g>
  <g>
    <circle
      className="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      r="15%"
      width="50%" />
    <text
      className="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%">
      comfort
    </text>
  </g>
  <g
    id="users">
    <g
      className="user-group">
      <rect
        className="in-focus"
        height="90"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        width="800"
        x="0"
        y={60} />
      <text
        className="username"
        x="60"
        y={60}>
        Adam Hall
      </text>
    </g>
    <g
      className="user-group">
      <rect
        className="not-in-focus"
        height="90"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        width="800"
        x="0"
        y={150} />
      <text
        className="username"
        x="60"
        y={150}>
        Caroline Hall
      </text>
    </g>
  </g>
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g>
    <rect
      class="not-in-focus"
      height="100%"
      id="chaos"
      width="100%"
    >
    </rect>
    <text
      class="area-label"
      id="label-chaos"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20"
    >
      chaos
    </text>
  </g>
  <g>
    <circle
      class="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      r="33%"
      width="50%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="20%"
    >
      stretch
    </text>
  </g>
  <g>
    <circle
      class="not-in-focus"
      cx="50%"
      cy="50%"
      height="50%"
      id="stretch"
      r="15%"
      width="50%"
    >
    </circle>
    <text
      class="area-label"
      id="label-stretch"
      text-anchor="middle"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      comfort
    </text>
  </g>
  <g id="users">
    <g class="user-group">
      <rect
        class="in-focus"
        height="90"
        width="800"
        x="0"
      >
      </rect>
      <text
        class="username"
        x="60"
      >
        Adam Hall
      </text>
    </g>
    <g class="user-group">
      <rect
        class="not-in-focus"
        height="90"
        width="800"
        x="0"
      >
      </rect>
      <text
        class="username"
        x="60"
      >
        Caroline Hall
      </text>
    </g>
  </g>
</svg>
`;
